From 61443fe8723872f5ac84ca44ce7e89451d7bdfeb Mon Sep 17 00:00:00 2001
From: Rainer Gerhards <rgerhards@adiscon.com>
Date: Fri, 9 Nov 2018 15:04:28 +0100
Subject: [PATCH] bugfix imfile: rsyslog re-sends data for files larger 2GiB

This occurs always if and only if
- reopenOnTruncate="" is set
- file grows over 2GiB in size

Then, the data is continously re-sent until the file becomes smaller
2GiB (due to truncation) or is deleted.

It is a regression introduced by 2d15cbc8221e385c5aa821e4a851d7498ed81850

closes https://github.com/rsyslog/rsyslog/issues/3249
---
 runtime/stream.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/runtime/stream.c b/runtime/stream.c
index 49005124d..4c631a814 100644
--- a/runtime/stream.c
+++ b/runtime/stream.c
@@ -674,7 +674,7 @@ static rsRetVal ATTR_NONNULL()
 checkTruncation(strm_t *const pThis)
 {
 	DEFiRet;
-	int ret;
+	off64_t ret;
 	off64_t backseek;
 	assert(pThis->bReopenOnTruncate);
 
